### PR TITLE
Add SkColorTable setting support to C API

### PR DESCRIPTION
**Pull request by Tylerflick**
This PR provides a getter as well as setter support for modifying the underlying color table of index8 type bitmaps.

### CODE REVIEW

**Comment by mattleibow** [comment]
@Tylerflick, this is good!

Just a suggestion, maybe it might be better to expose the `sk_bitmap_set_colortable` method as `sk_bitmap_set_pixels`, and have the parameters be `void*` and `sk_colortable_t*`. This way the same method can be used later if we ever want to actually swap out the pixels.

Then, in managed code we can write:

``` csharp
// we may not want this as NULL is a valid CT
void SetPixels(IntPtr pixels) {
    SetPixels(pixels, this.ColorTable);
}
void SetPixels(IntPtr pixels, SKColorTable ct = null) {
    sk_bitmap_set_pixels(pixels, ct != null ? ct.Handle : IntPtr.Zero);
}
void SetColorTable(SKColorTable ct) {
    SetPixels(this.GetPixels(), ct);
}
```

According to the docs above the `setPixels` method:

> Use this to assign a new pixel address for an existing bitmap. This will automatically release any pixelref previously installed. Only call this if you are handling ownership/lifetime of the pixel memory.

Are you able to confirm that this doesn't free the pixels first? I see that basically new SKBitmap creates the pixel data via a pixelref. This allocates the pixel data, and hooks up a "destructor" for when the ref gets to zero. Since we are freeing the pixel data, it will unref the pixelref. This may cause the pixels to be destroyed. This is just from looking at the code, so I can't say that this is what happens in real life.

**Comment by Tylerflick** [comment]
Good catch on the ref count. I've updated the function signature to match your suggestion.

I'll do some testing on the memory implications of this and whether or not a copy of the underlying pixel data needs to be made.

File: src/c/sk_bitmap.cpp [hunk 1]
```diff
@@ -300,3 +320,9 @@ sk_colortable_t* sk_bitmap_get_colortable(sk_bitmap_t* cbitmap)
 {
     return ToColorTable(AsBitmap(cbitmap)->getColorTable());
 }
+
+void sk_bitmap_set_pixels(sk_bitmap_t* cbitmap, void* pixels, sk_colortable_t* ctable)
+{
+    SkBitmap* bmp AsBitmap(cbitmap);
```
Comment (mattleibow) [inline src/c/sk_bitmap.cpp]: I think there is something wrong here. no equals (=) and this is a dupe of the function above...

Comment (Tylerflick) [reply src/c/sk_bitmap.cpp]: Oops. I didn't mean to push these changes to my public repo. I'm still debugging a few issues with the bindings.

**Comment by mattleibow** [comment]
Is this ready to go? Have you gotten around to testing whether the `setPixels` doesn't nuke the pixel data?

**Comment by Tylerflick** [comment]
I haven't found any issues in my testing. 👍

That being said, Skia is not checking as to whether the index of a set pixel is within the bounds of the color table, but it wasn't blowing up either.

**Pull request merged**